sql/core/src/main/java/org/apache/spark/sql/execution/UnsafeKVExternalSorter.java

```diff
@@ -141,9 +142,10 @@ public UnsafeKVExternalSorter(
 
         // Get encoded memory address
         // baseObject + baseOffset point to the beginning of the key data in the map, but that
-        // the KV-pair's length data is stored in the word immediately before that address
+        // the KV-pair's length data is stored at 2 * uaoSize bytes immediately before that address
```

Comment:
sorry I don't get it here why the address is related to uaoSize?

Reply:
The record format is:

 ```
(total length) (key length) (key) (value) (8 byte pointer to next value)
       |             |
    uaoSize       uaoSize
```

And we now get keyOffset, so we need back walk for 2 usao sizes to get the address.

Reply:
so you mean we need to add back 2 * uaoSize to cover the space of storing total length and key length? If that's the case then it makes sense.

Reply:
yes.